modules/SofaSparseSolver/src/SofaSparseSolver/SparseLUSolver.inl

```diff
@@ -85,12 +91,66 @@ void SparseLUSolver<TMatrix,TVector,TThreadManager>::invert(Matrix& M)
     invertData->A.nz = -1;							// # of entries in triplet matrix, -1 for compressed-col
     cs_dropzeros( &invertData->A );
 
+    invertData->perm.resize(invertData->A.n);
+    invertData->iperm.resize(invertData->A.n);
+
     invertData->tmp = (Real *) cs_malloc (invertData->A.n, sizeof (Real)) ;
-    invertData->S = cs_sqr (&invertData->A, order, 0) ;		/* ordering and symbolic analysis */
-    invertData->N = cs_lu (&invertData->A, invertData->S, f_tol.getValue()) ;		/* numeric LU factorization */
+
+    if( invertData->computePermutation){
+        fill_reducing_perm(invertData->A, invertData->perm.data(), invertData->iperm.data() ); // compute the fill reducing permutation
+        invertData->computePermutation = false;
```

Comment:
Same remark than for Cholesky. Cases where permutation must be recomputed must be supported